### DISCUSSION TITLE
Query not using join

### DISCUSSION THREAD

**Discussion by paymog** [prisma/prisma#22744] [Q&A] [answered]
### Question

I have the following prisma query which I'm executing against a Postgres database. I'm using the typescript sdk version 5.2.0.

```
  const projectsAndSubgraphs = await prisma.project.findMany({
    select: {
      id: true,
      billing_subscription_id: true,
      subgraph_deployments: {
        select: {
          subgraph_deployment_id: true,
        },
        where: {
          subgraph_deployment_id: {
            notIn: communitySubgraphs,
          },
          status: SubgraphStatus.Active,
        },
      },
      default_graph_node_cluster: {
        select: {
          name: true,
        },
      },
    },
    where: {
      billing_subscription_id: {
        startsWith: "sub_",
      },
    },
  });
```

I would expect this query to use `JOIN` clauses at the database layer but when I print the prisma queries being issued I see the following three queries

```
[12:14:00 UTC] INFO: prisma query
    query: "SELECT \"goldsky_api\".\"Project\".\"id\", \"goldsky_api\".\"Project\".\"billing_subscription_id\", \"goldsky_api\".\"Project\".\"default_graph_node_cluster_id\" FROM \"goldsky_api\".\"Project\" WHERE \"goldsky_api\".\"Project\".\"billing_subscription_id\"::text LIKE $1 OFFSET $2"
    params: "[\"sub_%\",0]"
[12:14:00 UTC] INFO: prisma query
    query: "SELECT \"goldsky_api\".\"SubgraphDeployment\".\"id\", \"goldsky_api\".\"SubgraphDeployment\".\"subgraph_deployment_id\", \"goldsky_api\".\"SubgraphDeployment\".\"project_id\" FROM \"goldsky_api\".\"SubgraphDeployment\" WHERE ((1=1 AND \"goldsky_api\".\"SubgraphDeployment\".\"status\" = $1) AND \"goldsky_api\".\"SubgraphDeployment\".\"project_id\" IN ($2)) OFFSET $3"
    params: "[\"Active\",\"project_clrovyyq7000afkp2cye59t5l\",0]"
[12:14:00 UTC] INFO: prisma query
    query: "SELECT \"goldsky_api\".\"GraphNodeCluster\".\"id\", \"goldsky_api\".\"GraphNodeCluster\".\"name\" FROM \"goldsky_api\".\"GraphNodeCluster\" WHERE \"goldsky_api\".\"GraphNodeCluster\".\"id\" IN ($1) OFFSET $2"
    params: "[\"shared\",0]"
```

It seems that the results are being joined at the application layer. Why is that happening? Is there a way to force Prisma to do database level `JOIN`s for this query?

### Expected behavior (optional)

I expect prisma to do database level joins and not perform application level joins for this particular query.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment, marked as answer]
Hey @paymog 👋 

You are looking for the `relationJoins` preview feature which we added in Prisma version 5.7.0. Can you enable it and check?
- https://github.com/prisma/prisma/releases/tag/5.7.0

The release notes above mention how you can enable it.

**Reply by paymog** [reply]
oh wow, I always assumed `JOIN`s were the default - turns out I was wrong. Turns out I was wrong.

**Comment by ludralph** [comment]
Hi @paymog 👋 

Thank you for raising this question.

We released support for `JOINS` preview feature in Prisma version [5.7.0](https://github.com/prisma/prisma/releases/tag/5.7.0) and have continued to ship improvements in version [5.8.0](https://github.com/prisma/prisma/releases/tag/5.8.0) which allows you to decide the  the strategy used to fetch relational data per query when the Preview feature is enabled. This will enable you to choose the most efficient strategy (either query or joins) for fetching relation data depending on your use case. You will need to update to the latest version of Prisma in order to use JOINS.

To get started using JOINs, enable the Preview feature in your Prisma schema:

// schema.prisma
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["relationJoins"]
}
```
Run prisma generate to regenerate Prisma Client and enable the Preview feature. Prisma Client will use a JOIN in your query to fetch relation data for a majority of the cases.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
